mapflow/dialogs/main_dialog.py

```diff
@@ -416,3 +423,52 @@ def setup_default_project(self, is_default: bool):
         self.deleteProject.setToolTip(tooltip)
         self.updateProject.setToolTip(tooltip)
 
+    def disable_shared_project(self, user_role: UserRole):
+        """Disable buttons depending on user role in a shared project.
+        Does not handle:
+        - processing renaming option that's inside saveOptionsButton (it's a QMenu(), so to not pass it to dlg, this option should be 
+                                                                      removed/added there the menu was initialized - in mapflow.py);
+        - deleteProject and updateProject (buttons instantly turn back on in 'on_project_change' in mapflow.py,
+                                           so it should be disabled there).
+
+        :param user_role: current user's role in a project (from ShareProject schema).
+        """        
+        if not user_role.can_start_processing:
+            self.disable_processing_start(reason=self.tr(f'Not enougth rights to start processing in a shared project ({user_role})'),
+                                                         clear_area=True,
+                                                         user_role=user_role)
+            # Disable widgets in a left panel
+            for i in range(self.procesingControlLayout.count()):
```

Comment:
It doesn't look good:
1. It heavily relies on the particular layout
2. In the same time, this dependency is implicit (we don't operate the layout in the application, so other developer may change the layout and expect the behavior to stay the same)
3. It is very verbose.

I thought that there is some container widget that contains all the processing control, bit it seems there is none, and the layout can't be operated as a container, so it complicates things.
What I would suggest:
1. Add property to the MainWindow class which will return a list of controls for processing, which in the function we will explicitly list (maybe even more lines, but explicit listing)
2. In this function use this list:
```
def enable_shared_project(self, user_role: UserRole):
    for control in self.project_controls:
        control.enable(user_role.can_start_processing)
```
and the same for review/delete/etc. What do you think?